### PR TITLE
Remove expect

### PR DESCRIPTION
**Pull request by dvdplm**
Fixes issue #8222

### CODE REVIEW

**Comment by ascjones** [comment]
Is returning `Ok(None)` the right thing in the case of a `DecoderError`? So the key exists but the data is invalid. I think we are losing some information there.

**Comment by dvdplm** [comment]
> I think we are losing some information there.

Do you think I should dig deeper into this and see if I can make `DecoderErrors` bubble up as the boxed trie error?

**Comment by ascjones** [comment]
That would be the ideal since this fn is already returning a Result, but if I recall it wasn't straightforward (can't remember why). 

Stepping back, we are decoding the Node RLP that was [self encoded](https://github.com/paritytech/parity/blob/master/util/patricia_trie/src/triedbmut.rs#L132). So again it's a case of either a corrupted DB or the node RLP encoding has changed and the DB wasn't migrated. In this exceptional scenario I think panicking would be better than returning `Ok(None)`. 

But if we can add the Result case and it is handled correctly further up then that would be best.

**Comment by dvdplm** [comment]
> That would be the ideal since this fn is already returning a Result, but if I recall it wasn't straightforward (can't remember why).

Attempted this in https://github.com/paritytech/parity/pull/8536/commits/1d54fc973aa6db2a8586bf55f0536cc11696bb7a – do you mind taking a look? I am struggling to write a decent test for it though (not sure how to make the look up fail with a decode error!).

File: util/patricia_trie/src/lookup.rs [hunk 1]
```diff
@@ -55,29 +55,32 @@ impl<'a, Q: Query> Lookup<'a, Q> {
...
-							false => None,
-						})
-					}
-					Node::Extension(slice, item) => {
-						if key.starts_with(&slice) {
-							node_data = item;
-							key = key.mid(slice.len());
-						} else {
-							return Ok(None)
+				match Node::decoded(node_data) {
```
Comment (ascjones) [inline util/patricia_trie/src/lookup.rs]: I think this nested match could be avoided with a `?` here and change the `From` impl to `Box<TrieError>`?

Comment (dvdplm) [reply util/patricia_trie/src/lookup.rs]: Good call, totally works! :)

**Comment by ascjones** [comment]
I did the same when I first started but you need two reviewers before merging.

**Pull request merged**